### DISCUSSION TITLE
Serving lots of TIF from a Google Storage bucket

### DISCUSSION THREAD

**Discussion by iibrahimli** [developmentseed/titiler#708] [Q&A] [answered]
I have the following:
 - A Google Storage bucket with a lot of TIF images (each about 2-5 MB)
 - A TiTiler instance that has access to the bucket

I want to use the TiTiler to serve the TIFs. I can request a single TIF using a URL like `gs://bucket-name/path/to/the/image.tif`, and e.g. I can view the TIF using the `/cog/map` endpoint. However, I have lots of TIFs in lots of folders, and it is impractical to add all of them separately. Is there a way to serve all of those files, what do I need to do, ideally without deploying any other software?

**Comment by vincentsarago** [comment, marked as answer]
You have 3 solutions:
- GDAL VRT: create a `.vrt` of your files and add external overviews (You'll need to set `GDAL_DISABLE_READDIR_ON_OPEN=FALSE` to read the external overview).
- Create a [MosaicJSON](https://github.com/developmentseed/cogeo-mosaic) file and use the `/mosaic` endpoints if you are using default `titiler.application` endpoints 
- Create STAC Items for all of you COGs and then use [`titiler.pgstac`](https://github.com/stac-utils/titiler-pgstac) (You'll need to set up a PgSTAC Database
